--- a/server/src/main/java/io/crate/planner/node/management/ExplainPlan.java
+++ b/server/src/main/java/io/crate/planner/node/management/ExplainPlan.java
@@ -120,16 +120,27 @@
-        if (context != null) {
-            assert subPlan instanceof LogicalPlan : "subPlan must be a LogicalPlan";
-            LogicalPlan plan = (LogicalPlan) subPlan;
-            var ramAccounting = ConcurrentRamAccounting.forCircuitBreaker(
-                "multi-phase",
-                dependencies.circuitBreaker(HierarchyCircuitBreakerService.QUERY),
-                plannerContext.transactionContext().sessionSettings().memoryLimitInBytes()
-            );
-            consumer.completionFuture().whenComplete((_, _) -> ramAccounting.close());
-            executePlan(
-                plan,
-                dependencies,
-                plannerContext,
-                consumer,
-                params,
-                subQueryResults,
-                new IdentityHashMap<>(),
-                new ArrayList<>(plan.dependencies().size()),
-                ramAccounting,
-                null
-            );
-        } else {
+        if (context != null) {
+            assert subPlan instanceof LogicalPlan : "subPlan must be a LogicalPlan";
+            LogicalPlan plan = (LogicalPlan) subPlan;
+            var ramAccounting = ConcurrentRamAccounting.forCircuitBreaker(
+                "multi-phase",
+                dependencies.circuitBreaker(HierarchyCircuitBreakerService.QUERY),
+                plannerContext.transactionContext().sessionSettings().memoryLimitInBytes()
+            );
+            consumer.completionFuture().whenComplete((__, ___) -> ramAccounting.close());
+            executePlan(
+                plan,
+                dependencies,
+                plannerContext,
+                consumer,
+                params,
+                subQueryResults,
+                new IdentityHashMap<>(),
+                new ArrayList<>(plan.dependencies().size()),
+                ramAccounting,
+                null
+            );
+        } else {
             if (subPlan instanceof LogicalPlan logicalPlan) {
                 if (verbose) {
                     List<RowN> rows = optimizerSteps.stream()
                         .map(step -> {
                             if (step.isInitial()) {
                                 return new RowN("Initial logical plan", step.planAsString());
                             }
                             return new RowN(step.rule().sessionSettingName(), step.planAsString());
                         })
                         .collect(Collectors.toCollection(ArrayList::new));
                     rows.add(new RowN("Final logical plan", printLogicalPlan(logicalPlan, plannerContext, showCosts)));
                     consumer.accept(InMemoryBatchIterator.of(rows, SENTINEL, false), null);
                 } else {
                     var planAsString = printLogicalPlan(logicalPlan, plannerContext, showCosts);
                     consumer.accept(InMemoryBatchIterator.of(new Row1(planAsString), SENTINEL), null);
                 }
             } else if (subPlan instanceof CopyFromPlan copyFromPlan && !verbose) {
                 BoundCopyFrom boundCopyFrom = CopyFromPlan.bind(
                     copyFromPlan.copyFrom(),
                     plannerContext.transactionContext(),
                     plannerContext.nodeContext(),
                     params,
                     subQueryResults);
                 ExecutionPlan executionPlan = CopyFromPlan.planCopyFromExecution(
                     copyFromPlan.copyFrom(),
                     boundCopyFrom,
                     dependencies.clusterService().state().nodes(),
                     plannerContext);
                 String planAsJson = DataTypes.STRING.implicitCast(PlanPrinter.objectMap(executionPlan));
                 consumer.accept(InMemoryBatchIterator.of(new Row1(planAsJson), SENTINEL), null);
             } else if (verbose) {
                 consumer.accept(null,
                     new UnsupportedOperationException("EXPLAIN VERBOSE not supported for " + subPlan.getClass().getSimpleName()));
             } else {
                 consumer.accept(InMemoryBatchIterator.of(
                     new Row1("EXPLAIN not supported for " + subPlan.getClass().getSimpleName()), SENTINEL), null);
             }
         }
